doc/unicode.md

```diff
@@ -0,0 +1,164 @@
+with older sources not using UTF-8.
+
+Read on for the gory details.
+## Alire assumptions and configuration
+
+By default, any crate initialized via `alr init` will have the `-gnatW8` switch
+in its build configuration, which presumes UTF-8 encoding of sources, and
+tweaks some internals of the compiler accordingly.
+
+This means that source files **must** use UTF-8 encoding when not using plain
+ASCII.
+
+Since GNAT compiles specifications and generic bodies in the context of the
+client project, once internalization is enabled in some parts of a build, it
+becomes necessary for all parts of the build to use the same `-gnatW8` setting.
+Otherwise, a file containing non-ASCII literals could be interpreted
+differently depending on the compilation context (as a standalone library or
+from a client of such a library).
+
+Most old GNAT projects are likely not to have `-gnatW8` enabled (although an
+UTF-8 file with BOM marker will have the same effect). For crates that do not
+contain string literals outside of ASCII or engage on I/O of such strings, this
+should not make any difference. For those for which internationalization
+matters, however, there is no sensible way forward but to embrace Unicode, with
+UTF-8 being the standard encoding nowadays for files and terminals.
+
+This means that a certain amount of breakage might happen for 'legacy'
+libraries not yet adapted to `-gnatW8`. From the Alire project we are trying to
+get in front of this problem with early detection of such libraries in the
+Alire ecosystem, and universally using `-gnatW8` from version 2.0 on.
+
+## Recommendations
+
+You can use a library designed to shield you from Unicode details, such as
+`vss` or `uxstrings`.
+
+Otherwise, these recommendations should keep you safe:
+
+1. Use `-gnatW8` to compile (this is Alire's default) and save your sources in UTF-8 encoding.
+1. Use Wide_Wide_String to store any text that may exceed the ASCII character set.
+    1. Alternatively, use `Ada.Strings.UTF_Encoding.Wide_Wide_Strings.Encode` to get UTF-8 strings from Unicode literals.
+1. Convert if necessary at the time of output to the desired encoding for display/storage.
+    1. By default, Ada.Wide_Wide_Text_IO will use UTF-8 encoding if `-gnatW8` is in effect.
+    1. Most modern plain text formats expect UTF-8 encoding.
+    1. On Linux terminals this is the usual default.
+    1. On Windows 10:
+        1. You can enable UTF-8 once by running `chcp 65001` in your terminal.
+        1. You can enable UTF-8 in terminals system-wide.
+            - This will break programs assuming the default Windows 10 encoding.
+    1. On Windows 11, UTF-8 terminals are the default.
+1. You can use `GNAT.IO` or `Ada.Streams.Stream_IO` to output properly encoded strings, as they don't manipulate the bytes.
+1. You **cannot** use `Ada.Text_IO` to output UTF-8-encoded `String` variables, as Latin-1 encoding is expected.
```

Comment:
That sounds scary since every Ada will ask you to use `Ada.Text_IO` to print strings.

Reply:
Yeah, it's the ugliest aspect of this issue. Unfortunately this works "by chance" when the terminal is UTF-8, the file is UTF-8, but no `-gnatW8` is given, as TIO lets "fake" Latin-1 strings which are UTF-8 get by unmodified. If the source is properly Latin-1-encoded then the output of TIO is again wrong.